### PR TITLE
feat(mocks-in-e2e-tests): example app with various e2e test approach

### PR DESCRIPTION
**Pull request by kgajowy**

### CODE REVIEW

File: articles/mocks-in-e2e-tests/src/app.module.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import {Module} from '@nestjs/common';
```
Comment (Sikora00) [inline articles/mocks-in-e2e-tests/src/app.module.ts]: prettier

File: articles/mocks-in-e2e-tests/src/orders/service.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      uuid,
+    }).toPromise();
+  }
+
+  async getOrderStatus(
+    uuid: string,
+    /**
+     * for brevity, skip real configs
+     */
+    paymentServiceUrl = process.env.PAYMENT_GATEWAY ?? ""
```
Comment (Sikora00) [inline articles/mocks-in-e2e-tests/src/orders/service.ts]: maybe as a class property?

File: articles/mocks-in-e2e-tests/test/payment-gateway-fake/bootstrap.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+let app: INestApplication | undefined;
+
+export const start = async (port: number): Promise<string> => {
+  if (app) {
+    return app.getUrl();
+  }
+  app = await NestFactory.create(ProviderModule);
+  await app?.init();
+  await app.listen(port);
```
Comment (Sikora00) [inline articles/mocks-in-e2e-tests/test/payment-gateway-fake/bootstrap.ts:13]: once with a question mark and once not?

**Review by Sikora00 — CHANGES_REQUESTED** [review]

**Comment by kgajowy** [comment]
@MaciejSikorski re:

**Review by Sikora00 — APPROVED** [review]

**Pull request merged**